warn/scrapers/mi.py

```diff
@@ -91,6 +107,185 @@ def scrape(
+    """Process 2007-2015 pdfs."""
+    # split at newline
+    txt = txt.split("\n")
+    ans = []
+    for i in txt:
+        final = []
+        # split at spaces, which are used as column divders
+        raw = i.split(" ")
+        # remove space placeholders
+        final = [j.replace("\xa0", " ").replace("\xad", " ") for j in raw]
+
+        # removing edge cases such as the "Notes:" section and blank lines
+        if (
+            len(final) == 5
+            and (final[0] not in ["Company", "Company Name"])
+            and final[1] != "Received"
+            and final != [" ", " ", " ", " ", " "]
+        ):
+            ans.append(final)
+        # accounting for the 2008-2009 minor format changes
+        if len(final) == 6:
+            if (final[2][-4:] == "2008" or final[2][-4:] == "2009") and (
+                final[0] != "January 1 through December 31:"
+            ):
+                final.pop(-1)
+                ans.append(final)
+
+    return ans
+
+
+# there are around 15 lines with unique spacing issues
+# there doesn't seem to be a pattern to them,
+# so I just fixed them manually
+# there may be a more elegant solution
+def handle_edge_cases(row):
+    """Handle edge cases in 2000-2006 pdfs."""
+    if row[0] == "General Motors Nao Orion AsseOrion":
+        row[1] = "Orion"
+        row[0] = row[0][-5:]
+    if row[1] == "LifDetroit":
+        row[1] = "Detroit"
+    if row[1] == "ILapeer":
+        row[1] = "Lapeer"
+    if row[0] == "Standard Federal Bank #320     Troy":
+        row[1] = "Troy"
+        row[0] = "Standard Federal Bank #320"
+    if row[0] == "Standard Federal Bank #340     Troy":
+        row[1] = "Troy"
+        row[0] = "Standard Federal Bank #340"
+    if row[1] == "IFlint":
+        row[1] = "Flint"
+    if row[1] == "DivisioJackson":
+        row[1] = "Jackson"
+    if row[1] == "IncWyoming":
+        row[1] = "Wyoming"
+    if row[0] == "Michigan Machine And EngineerFenton":
+        row[1] = "Fenton"
+        row[0] = row[0][-6:]
+    if row[1] == "CompDearborn Heights":
+        row[1] = "Dearborn Heights"
+    if row[1] == "InPetersburg":
+        row[1] = "Petersburg"
+    if row[0] == "Asplundh Tree Company #93     Wixom":
+        row[0] = row[0][-5:]
+        row[1] = "Wixom"
+    if row[0] == "Asplundh Tree Company #43     Mt.":
+        row[0] = "Asplundh Tree Company #43"
+        row[1] = "Mt. Clemens"
+    if row[0] == "Asplundh Tree Company #843   Howell":
+        row[0] = "Asplundh Tree Company #843"
+        row[1] = "Howell"
+    if row[1] == "RegionaSaginaw":
+        row[1] = "Saginaw"
+    if row[0] == "Robert Bosch Corp. Chassis DivisSi":
+        row[0] = "Robert Bosch Corp. Chassis Division"
+        row[1] = "St. Joseph"
+    if row[0] == "Magna / Cosma Body & Chassis Troy":
+        row[0] = row[0][-4:]
+        row[1] = "Troy"
+    if row[1] == "Corp.Flint":
+        row[1] = "Flint"
+    if row[1] == "Inc.Howell":
+        row[1] = "Howell"
+    return row
```

Comment:
We don't need to do this kind of clean up in the scraper. It can be handled downstream. Let's cut it.

Reply:
Done!